src/lib/ledger_catchup/normal_catchup.ml

```diff
@@ -680,38 +680,52 @@ let run ~logger ~precomputed_values ~trust_system ~verifier ~network ~frontier
                               , `List
                                   (List.map errors ~f:(fun err ->
                                        `String (display_error err) )) ) ] ;
-                        if contains_no_common_ancestor errors then (
-                          let transitions =
-                            List.concat_map subtrees ~f:Rose_tree.flatten
-                          in
-                          List.iter transitions ~f:(fun cached_transition ->
-                              let transition =
-                                Cached.peek cached_transition
+                        if contains_no_common_ancestor errors then
+                          List.iter subtrees ~f:(fun subtree ->
+                              let root_transition =
+                                Rose_tree.root subtree |> Cached.peek
                                 |> Envelope.Incoming.data
                               in
-                              let transition_hash =
-                                External_transition.Initial_validated
-                                .state_hash transition
+                              let children_transitions =
+                                List.concat_map
+                                  (Rose_tree.children subtree)
+                                  ~f:Rose_tree.flatten
+                              in
+                              let children_state_hashes =
+                                List.map children_transitions
+                                  ~f:(fun cached_transition ->
+                                    Cached.peek cached_transition
+                                    |> Envelope.Incoming.data
+                                    |> External_transition.Initial_validated
+                                       .state_hash )
                               in
                               [%log error]
                                 ~metadata:
-                                  [ ( "state_hash"
-                                    , State_hash.to_yojson transition_hash )
+                                  [ ( "state_hashes_of_children"
+                                    , `List
+                                        (List.map children_state_hashes
+                                           ~f:State_hash.to_yojson) )
+                                  ; ( "state_hash_of_root"
```

Comment:
nit: use `state_hash` (as specified in the error message `$state_hash`)  and `protocol_state`? might confuse with the root of the frontier